### PR TITLE
ikmsg: fix EPIPE error check in the /dev/kmsg reader loop

### PR DESCRIPTION
**Pull request by hsaliak**
This is a patch for the issue reported in  #4395. 

kmsg is a unique device, which can recover from EPIPE errors. The code in the module checks for this, but the return value from the libc read call always returns -1, and sets the appropriate errno.

Before this patch, the code checks the return value of the read() call against -EPIPE, which never happens. 

This meant that when an EPIPE error actually happens, the fd  is set to -1 and indefinitely retried. rsyslog does not recover from this. Even though the `for` loop in `readikmsg` is broken out of, this function is repeatedly invoked in a `while` loop.

Note: there is an additional improvement possible - `readikmsg` needs better error checking on the fd. I suspect that this was rarely an issue because /dev/kmsg goes truly invalid when the system is actually shutting down.

The fix here is to check the return value as well as the errno. This will allow the intended logic in the code to happen, and for the reads to recover after a `read` on `/dev/kmsg` returns an `EPIPE`.

closes https://github.com/rsyslog/rsyslog/issues/4395

### CODE REVIEW

**Comment by hsaliak** [comment]
@rgerhards it seems that as a first time contributer, I would need a maintainer to approve running the workflows. Would you be able to help?

**Comment by rgerhards** [comment]
> it seems that as a first time contributor, I would need a maintainer to approve running the workflows.

Done! Thx for the PR!

**Review by rgerhards — APPROVED** [review]

**Comment by apolarus** [comment]
@hsaliak we saw this issue again yesterday on [DTP-18609](https://jira.int.aurora.tech/browse/DTP-18609) running  [AVDeployThud-2022.0119.52](https://shipyard.common.cloud.aurora.tech/release/AVDeployThud-2022.0119.52). Would this still be expected? I'm not clear on how the flow of _rsyslog:master_ should look.

**Comment by rgerhards** [comment]
Well, that depends on when you build from master (or however you access rsyslog code). After merge, it's in the main branch (still called "master" because of some automation dependencies which require quite a bit of work).

**Pull request merged**
